### PR TITLE
THU/cypress_stubbing

### PR DESCRIPTION
**Pull request by csm-thu**
**TLDR of this PR**:
- add fixtures to use as default data when stubbing is enabled (not all of these fixtures are used yet)
- add generic functions to make stubbing straightforward when writing tests:
  - _stubbing service_ is a class to keep the current state of stubbed data when tests are running
  - _api utils_ implement requests interception functions behavior (with conditions based on whether or not the stubbing is active)
  - _auth utils_ implement function to manipulate JWT tokens and be able to read & modify user roles on login
  - _route utils_ offer wrappers for routing functions (to prevent duplicating code with interceptions)
- some existing actions & tests have been adapted to be compatible with stubbing
- 2 new tests using stubbing: _NoScenario_ and _UserApplicationRoles_
- 1 existing test modified to use stubbing: _ScenarioListOrder_
- all other existing tests are not using stubbing yet

### CODE REVIEW

File: cypress/fixtures/stubbing/default/solution.js [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    {
+      ...SOLUTION_PARAMETER_EXAMPLE,
+      id: 'start_date',
+      labels: { fr: 'Date de départ', en: 'Start date' },
+      varType: 'date',
+    },
+    {
+      ...SOLUTION_PARAMETER_EXAMPLE,
+      id: 'additional_seats',
+      labels: { fr: 'Sièges additionnels', en: 'Additionnal seats' },
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply cypress/fixtures/stubbing/default/solution.js:112]: Nice catch ^^
